tx-pool/src/component/pool_map.rs

```diff
@@ -0,0 +1,530 @@
+//! Top-level Pool type, methods, and tests
+extern crate rustc_hash;
+extern crate slab;
+use crate::component::edges::Edges;
+use crate::component::entry::EvictKey;
+use crate::component::links::{Relation, TxLinksMap};
+use crate::component::score_key::AncestorsScoreSortKey;
+use crate::error::Reject;
+use crate::TxEntry;
+use ckb_logger::{debug, trace};
+use ckb_multi_index_map::MultiIndexMap;
+use ckb_types::core::error::OutPointError;
+use ckb_types::packed::OutPoint;
+use ckb_types::{
+    bytes::Bytes,
+    core::{cell::CellChecker, TransactionView},
+    packed::{Byte32, CellOutput, ProposalShortId},
+};
+use ckb_types::{
+    core::cell::{CellMetaBuilder, CellProvider, CellStatus},
+    prelude::*,
+};
+use std::collections::HashSet;
+
+use super::links::TxLinks;
+
+type ConflictEntry = (TxEntry, Reject);
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum Status {
+    Pending,
+    Gap,
+    Proposed,
+}
+
+#[derive(Copy, Clone)]
+enum EntryOp {
+    Add,
+    Remove,
+}
+
+#[derive(MultiIndexMap, Clone)]
+pub struct PoolEntry {
+    #[multi_index(hashed_unique)]
+    pub id: ProposalShortId,
+    #[multi_index(ordered_non_unique)]
+    pub score: AncestorsScoreSortKey,
+    #[multi_index(ordered_non_unique)]
```

Comment:
is it better to use `hashed_non_unique` for performance since we don't need to iterate the pool map by status?

Reply:
fixed, thanks.